quickwit/quickwit-aws/src/lib.rs

```diff
@@ -98,12 +99,21 @@ impl ExtendedChainProvider {
 #[async_trait]
 impl ProvideAwsCredentials for ExtendedChainProvider {
     async fn credentials(&self) -> Result<AwsCredentials, CredentialsError> {
-        if let Ok(credentials) = self.web_identity_provider.credentials().await {
-            return Ok(credentials);
-        }
-        if let Ok(credentials) = self.chain_provider.credentials().await {
-            return Ok(credentials);
+        let identity_provider = self.web_identity_provider.credentials().await;
+        let chain_provider = self.chain_provider.credentials().await;
```

Comment:
Why resolve the second one if the first one succeeded?

```rust
let identity_provider_error = match self.web_identity_provider.credentials().await {
  Ok(credentials) => return Ok(credentials),
  Err(error) => error,
};
let chain_provider_error = match self.chain_provider.credentials().await {
  Ok(credentials) => return Ok(credentials),
  Err(error) => error,
};

error!(...)
Err(...)
```

Reply:
nice, this is better indeed